be/src/formats/parquet/column_reader.cpp

```diff
@@ -134,28 +135,87 @@ Status ColumnDictFilterContext::rewrite_conjunct_ctxs_to_predicate(StoredColumnR
     return Status::OK();
 }
 
-void ColumnReader::get_subfield_pos_with_pruned_type(const ParquetField& field, const TypeDescriptor& col_type,
-                                                     bool case_sensitive, std::vector<int32_t>& pos) {
+void ColumnReader::get_subfield_pos_with_pruned_type(
+        const ParquetField& field, const TypeDescriptor& col_type, bool case_sensitive, std::vector<int32_t>& pos,
+        const TPhysicalSchemaField* physical_schema_field,
+        std::vector<const TPhysicalSchemaField*>& physical_schema_subfield_vec) {
     DCHECK(field.type.type == LogicalType::TYPE_STRUCT);
+    if (physical_schema_field != nullptr) {
+        std::unordered_map<std::string, const TPhysicalSchemaField*> logical_name_2_physical_field;
+        for (const auto& child : physical_schema_field->children) {
+            logical_name_2_physical_field.emplace(Utils::format_name(child.logical_name, case_sensitive), &child);
+        }
 
-    // build tmp mapping for ParquetField
-    std::unordered_map<std::string, size_t> field_name_2_pos;
-    for (size_t i = 0; i < field.children.size(); i++) {
-        const std::string format_field_name =
-                case_sensitive ? field.children[i].name : boost::algorithm::to_lower_copy(field.children[i].name);
-        field_name_2_pos.emplace(format_field_name, i);
-    }
+        std::unordered_map<int32_t, size_t> field_id_2_pos;
+        std::unordered_map<std::string, size_t> physical_name_2_pos;
```

Comment:
why not just use logical_name->pos map?

Reply:
we need to check the filed_id or physical name exist in parquet field